cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1482,7 +1482,23 @@ macro(build_flatbuffers)
 endmacro()
 
 if(ARROW_WITH_FLATBUFFERS)
-  resolve_dependency(Flatbuffers)
+  get_dependency_source(Flatbuffers)
+  if(Flatbuffers_SOURCE STREQUAL "AUTO")
+    find_package(Flatbuffers QUIET)
+    if(NOT Flatbuffers_FOUND)
+      find_package(FlatbuffersAlt)
```

Comment:
Perhaps comment on why this is required?

Reply:
Done